### PR TITLE
Kernel loops examples

### PR DESCRIPTION
**Pull request by fwyzard**
This PR uses the device functions introduced in #2369 to simplify some of the examples.

### CODE REVIEW

File: include/alpaka/exec/IndependentElements.hpp
Comment (psychocoderHPC) [inline include/alpaka/exec/IndependentElements.hpp]: I found that the inclusion of this header has no affect in C++ because C++ itself defines these operants.
@fwyzard Why do you added these header?

https://cplusplus.com/reference/ciso646/

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: include/alpaka/exec/IndependentElements.hpp
Comment (fwyzard) [reply include/alpaka/exec/IndependentElements.hpp]: Because MSVC is a stupid compiler and _does not_ define the keywords by default: https://learn.microsoft.com/en-us/cpp/build/reference/microsoft-extensions-to-c-and-cpp?view=msvc-170&redirectedfrom=MSDN#iso646h-not-enabled (and `/Ze` is the default).

Comment (fwyzard) [reply include/alpaka/exec/IndependentElements.hpp]: OK, MSVC _does_ define the keywords in c++20 mode: https://godbolt.org/z/eMre9PMvT .

So we can remove them again once we switch to c++20.

Comment (SimeonEhrig) [reply include/alpaka/exec/IndependentElements.hpp]: Please add the comment `// TODO: can be removed if C++ 20 is the minimum requirement`
This allows us to use `grep` for cleaning up the code, when we move to C++20.

**Review by SimeonEhrig — APPROVED** [review]

**Comment by SimeonEhrig** [comment]
@psychocoderHPC Your change request blocks the merge.

**Comment by fwyzard** [comment]
@SimeonEhrig are there some other tests that you would like to see migrated to the new loops ?

**Comment by SimeonEhrig** [comment]
> @SimeonEhrig are there some other tests that you would like to see migrated to the new loops ?

I have to check the examples. But would be cool if we migrate a multi dimensional kernel.

**Review by psychocoderHPC — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
> @psychocoderHPC Your change request blocks the merge.

you should be able to dismiss my review.

**Pull request merged**
